File: csharp/DesiArchitect.DesignPatterns/05_Singleton/Before/OrderService.cs
namespace DesiArchitect.DesignPatterns.Singleton.Before;

// Any class using this is bound to the static instance
public class OrderService
{
    public void PlaceOrder(string item)
    {
        var config = AppConfigStatic.Instance;   // hardcoded dependency
        Console.WriteLine($"  [Before] OrderService using AppConfig.Instance directly");
        Console.WriteLine($"     Placing order for {item} using key: {config.ApiKey}");
        // Cannot mock this in unit tests
        // Cannot swap config for testing environment
    }
}

File: typescript/src/05-singleton/before/order-service.ts
import { AppConfigStatic } from './app-config-static.js';

export class OrderService {
  placeOrder(item: string): void {
    const config = AppConfigStatic.instance;
    console.log('  [Before] OrderService using AppConfig.Instance directly');
    console.log(`     Placing order for ${item} using key: ${config.apiKey}`);
  }
}